### PR TITLE
Fix: Make the ANR Atomic flags immutable

### PR DESCRIPTION
**Pull request by marandaneto**
## :scroll: Description
Fix: Make the ANR Atomic flags immutable

## :bulb: Motivation and Context
https://stackoverflow.com/questions/66095728/sentry-android-anr-could-there-be-a-false-alarm/66097964?noredirect=1#comment116884716_66097964

## :green_heart: How did you test it?
could not reproduce the issue not being final, but the behavior is still the same.

## :pencil: Checklist

- [X] I reviewed the submitted code
- [ ] I added tests to verify the changes
- [ ] I updated the docs if needed
- [X] No breaking changes

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -5,6 +5,7 @@
 * Fix: Disable Gson HTML escaping
 * Enchancement: Support @SentrySpan and @SentryTransaction on classes and interfaces. (#1243)
 * Ref: Simplify RestTemplate instrumentation (#1246)
+* Fix: Make the ANR AtomicLong final
```
Comment (marandaneto) [inline CHANGELOG.md:8]:
```suggestion
* Fix: Make the ANR Atomic flags immutable
```

**Review by philipphofmann — APPROVED** [review]
LGTM, but these properties are now not immutable. They are final. So maybe we want to change this in the title and the changelog?

**Comment by marandaneto** [comment]
> LGTM, but these properties are now not immutable. They are final. So maybe we want to change this in the title and the changelog?

@philipphofmann not sure if I follow, they are immutable

**Comment by philipphofmann** [comment]
Immutable means that you can't change the value of the object. Final means you can't change the reference. Both `tick` and `reported` are now final, but you can still change their value by using their setters and other methods.

**Comment by marandaneto** [comment]
> Immutable means that you can't change the value of the object. Final means you can't change the reference. Both `tick` and `reported` are now final, but you can still change their value by using their setters and other methods.

ok so to clarify what I want to achieve is the immutable state of assigning new references to the ANR Atomic flags, that's why I meant `ANR Atomic flags` in the title/PR change and not the whole class or its internal fields (which are mutable but atomic).
what would you suggest as a title/changelog?

**Review by bruno-garcia — APPROVED** [review]
good catch!

**Pull request merged**